src/main/java/servlets/ReceiptAnalysis.java

```diff
@@ -161,10 +162,11 @@ private static String retrieveText(ByteString imageBytes)
 
       ClassifyTextResponse response = client.classifyText(request);
 
-      // TODO: Parse category strings into more natural categories
+      // Parse category strings into more natural categories
+      // e.g. "/Food & Drink/Restaurants" becomes "Food & Drink" and "Restaurants"
       categories = response.getCategoriesList()
                        .stream()
-                       .map(category -> category.getName())
+                       .flatMap(category -> Stream.of(category.getName().substring(1).split("/")))
```

Comment:
If a computation isn’t self-explanatory, I would consider moving the code to a helper method and use method reference instead.